tests/Aspire.Hosting.Azure.Tests/AzureCosmosDBExtensionsTests.cs

```diff
@@ -240,6 +247,120 @@ param principalId string
         Assert.Equal(expectedBicep, manifest.BicepText);
     }
 
+    [Fact]
+    public async Task AddAzureCosmosDatabase_WorksWithAccessKeyAuth_ChildResources()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create(DistributedApplicationOperation.Run);
+
+        var cosmos = builder.AddAzureCosmosDB("cosmos")
+            .WithAccessKeyAuthentication();
+        var database = cosmos.AddCosmosDatabase("db1");
+        var container = database.AddContainer("container1", "id");
+
+        builder.AddContainer("api", "myimage")
+            .WithReference(cosmos);
+
+        Assert.Equal("{cosmos.secretOutputs.connectionString}", cosmos.Resource.ConnectionStringExpression.ValueExpression);
+        Assert.Equal("{cosmos.secretOutputs.connectionString};Database=db1", database.Resource.ConnectionStringExpression.ValueExpression);
```

Comment:
This is not correct. We should be using the whole connection string from the secrets.